packages/xo-web/src/common/form/number.js

```diff
@@ -0,0 +1,47 @@
+import PropTypes from 'prop-types'
+import React from 'react'
+import { injectState, provideState } from '@julien-f/freactal'
+
+const Number = [
+  provideState({
+    effects: {
+      onChange: (_, { target: { value } }) => (state, props) => {
+        if (value === '') {
+          if (!props.optional) {
+            return
```

Comment:
This would prevent the user from deleting the last digit.

Reply:
It's expected, The value must be a number and it's required. He can decrease using arrows or replace it. The implementation is like `Editable#Number`.

Reply:
What do you think about replace it with the  `min` value default to 0 ?

Reply:
What if I typed 9999 when I actually wanted to type 1111 ? I want to delete all the 9s to start over and type 1111 but I can't.

Reply:
I agree, you should differentiate between the textual value of the input and the emitted numeric value of `Number`.